File: snippets/csharp/discover_points.cs
using Qdrant.Client;
using Qdrant.Client.Grpc;

var client = new QdrantClient("localhost", 6334);

await client.DiscoverAsync(
  collectionName: "{collection_name}",
  target: new TargetVector
  {
    Single = new VectorExample { Vector = new float[] { 0.2f, 0.1f, 0.9f, 0.7f }, }
  },
  context:
  [
    new()
    {
      Positive = new VectorExample { Id = 100 },
      Negative = new VectorExample { Id = 718 }
    },
    new()
    {
      Positive = new VectorExample { Id = 200 },
      Negative = new VectorExample { Id = 300 }
    }
  ],
  limit: 10
);

File: snippets/rust/discover_points.rs
async fn main() -> Result<(), qdrant_client::QdrantError> {
    use qdrant_client::qdrant::{
        target_vector::Target, vector_example::Example, ContextExamplePairBuilder,
        DiscoverPointsBuilder, VectorExample,
    };
    use qdrant_client::Qdrant;

    let client = Qdrant::from_url("http://localhost:6334").build()?;

    client
        .discover(
            DiscoverPointsBuilder::new(
                "{collection_name}",
                vec![
                    ContextExamplePairBuilder::default()
                        .positive(Example::Id(100.into()))
                        .negative(Example::Id(718.into()))
                        .build(),
                    ContextExamplePairBuilder::default()
                        .positive(Example::Id(200.into()))
                        .negative(Example::Id(300.into()))
                        .build(),
                ],
                10,
            )
            .target(Target::Single(VectorExample {
                example: Some(Example::Vector(vec![0.2, 0.1, 0.9, 0.7].into())),
            })),
        )
        .await?;

    Ok(())
}
